tls/s2n_connection.c

```diff
@@ -659,6 +653,13 @@ int s2n_connection_wipe(struct s2n_connection *conn)
         conn->actual_protocol_version = s2n_highest_protocol_version;
     }
 
+    /* Initialize remaining values */
+    conn->blinding = S2N_BUILT_IN_BLINDING;
+    conn->session_ticket_status = S2N_NO_TICKET;
+    /* Initialize the cookie stuffer with zero length. If a cookie extension
+     * is received, the stuffer will be resized according to the cookie length */
+    POSIX_GUARD(s2n_stuffer_growable_alloc(&conn->cookie_stuffer, 0));
```

Comment:
Is this a memory leak if there is a cookie? Do we free the cookie stuffer somewhere else? If we don't, we'll need to free it before this point.

Maybe a better model is to keep the alloc in s2n_connection_new and just call s2n_stuffer_resize here. Looks like that's what we do with most other stuffers.

Reply:
Other possible lifecycle: allocate the memory when the cookie extension is received (so in the extension's recv method). The stuffer wouldn't even need to be growable, bc we'd know the length. Then we would just need to free in wipe, if we're not already freeing elsewhere (probably safe to free in wipe anyway).

Reply:
done in https://github.com/aws/s2n-tls/pull/3282/commits/c1a5746f42ee0a33204fe82baa47338a407719f2

Reply:
Do you think we need to create an issue to track the cleaner solution of turning the cookie stuffer into a blob and allocating it with s2n_realloc in the extension recv method?

Reply:
Created issue https://github.com/aws/s2n-tls/issues/3287